### PR TITLE
Update CHANGELOG for Firestore v0.10.1

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

**Comment by morganchen12** [comment]
@paulb777 #745 is a different crash that's not related to network condition, but afaik is included in this release as well.

**Comment by wilhuff** [comment]
Yep. I'm collecting up the issue numbers and I'll annotate

**Comment by wilhuff** [comment]
Done. PTAL. The first fix has no issue in this repo (first detected on Android).

File: Firestore/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
 # Unreleased
+
+# v0.10.1
 - [fixed] Fixed a regression in Firebase iOS release 4.8.1 that could in certain
```
Comment (morganchen12) [inline Firestore/CHANGELOG.md]: nit: This should maybe refer to 4.8.2, since 4.8.1 was made unavailable following some critical analytics regressions.

Comment (paulb777) [reply Firestore/CHANGELOG.md]: I'd vote to leave as is. Regression introduced in 4.8.1, continued in 4.8.2 and fixed in 4.9.0

**Pull request merged**